pmd-java/src/main/java/net/sourceforge/pmd/lang/java/ast/AssignmentOp.java

```diff
@@ -0,0 +1,95 @@
+/**
+ * BSD-style license; for more info see http://pmd.sourceforge.net/license.html
+ */
+
+package net.sourceforge.pmd.lang.java.ast;
+
+import static net.sourceforge.pmd.lang.java.ast.BinaryOp.ADD;
+import static net.sourceforge.pmd.lang.java.ast.BinaryOp.AND;
+import static net.sourceforge.pmd.lang.java.ast.BinaryOp.DIV;
+import static net.sourceforge.pmd.lang.java.ast.BinaryOp.LEFT_SHIFT;
+import static net.sourceforge.pmd.lang.java.ast.BinaryOp.MUL;
+import static net.sourceforge.pmd.lang.java.ast.BinaryOp.OR;
+import static net.sourceforge.pmd.lang.java.ast.BinaryOp.RIGHT_SHIFT;
+import static net.sourceforge.pmd.lang.java.ast.BinaryOp.SUB;
+import static net.sourceforge.pmd.lang.java.ast.BinaryOp.UNSIGNED_RIGHT_SHIFT;
+import static net.sourceforge.pmd.lang.java.ast.BinaryOp.XOR;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.Map;
+import java.util.stream.Collectors;
+import javax.annotation.Nullable;
+
+
+/**
+ * An assignment operator for {@link ASTAssignmentExpression}.
+ *
+ * <pre class="grammar">
+ *
+ * AssignmentOp ::= "=" | "*=" | "/=" | "%=" | "+=" | "-=" | "&lt;&lt;=" | "&gt;&gt;=" | "&gt;&gt;&gt;=" | "&amp;=" | "^=" | "|="
+ *
+ * </pre>
+ *
+ * @see BinaryOp
+ * @see UnaryOp
+ */
+public enum AssignmentOp {
+    EQ("=", null),
+    AND_EQ("&=", AND),
+    OR_EQ("|=", OR),
+    XOR_EQ("^=", XOR),
+    ADD_EQ("+=", ADD),
+    SUB_EQ("-=", SUB),
+    MUL_EQ("*=", MUL),
+    DIV_EQ("/=", DIV),
+    LEFT_SHIFT_EQ("<<=", LEFT_SHIFT),
+    RIGHT_SHIFT_EQ(">>=", RIGHT_SHIFT),
+    UNSIGNED_RIGHT_SHIFT_EQ(">>>=", UNSIGNED_RIGHT_SHIFT);
+
+    private static final Map<String, AssignmentOp> LOOKUP = Collections.unmodifiableMap(
+        Arrays.stream(values()).collect(Collectors.toMap(Object::toString, op -> op))
```

Comment:
you sometimes use `Function.identity()` and sometimes do this…

Reply:
Well to me `Function.identity()` is faster to write with the autocomplete, but `op -> op` is more readable IMO. We should probably create a rule to enforce a consistent style